components/kyma-environment-broker/internal/orchestration/kyma/orchestration.go

```diff
@@ -0,0 +1,117 @@
+package kyma
+
+import (
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/orchestration"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type upgradeKymaOrchestration struct {
+	db                  storage.Orchestration
+	resolver            orchestration.RuntimeResolver
+	kymaUpgradeExecutor process.Executor
+	log                 logrus.FieldLogger
+}
+
+func NewUpgradeKymaOrchestration(db storage.Orchestration, kymaUpgradeExecutor process.Executor, resolver orchestration.RuntimeResolver, log logrus.FieldLogger) process.Executor {
```

Comment:
To be precise, we do not return `Orchestration` with this constructor, but more like a Manager kind of thing. We should change this type name to avoid confusion in domain-specific terms.